js/modules/k6/http/response.go

```diff
@@ -137,6 +138,36 @@ func (res *HTTPResponse) Json() goja.Value {
 		default:
 			common.Throw(common.GetRuntime(res.ctx), errors.New("Invalid response type"))
 		}
+
+		if hasSelector {
+			result := gjson.GetBytes(body, selector[0])
```

Comment:
Agree, but I think it's worth to put this method out of the selector section. 
To avoid parsing json at all. 
In this case, will retain integrity with previous behavior when not valid JSON raise the error.
Check my commit